consensus/service/src/attested_api_service.rs

```diff
@@ -10,31 +10,41 @@ use mc_common::{
     HashSet,
 };
 use mc_consensus_enclave::ConsensusEnclave;
-use mc_util_grpc::{rpc_logger, rpc_permissions_error, send_result};
+use mc_util_grpc::{auth::Authenticator, rpc_logger, rpc_permissions_error, send_result};
 use mc_util_metrics::SVC_COUNTERS;
 use std::sync::{Arc, Mutex};
 
 #[derive(Clone)]
-pub struct AttestedApiService<E: ConsensusEnclave + Send, S: Session> {
-    enclave: E,
+pub struct AttestedApiService<S: Session> {
+    enclave: Arc<dyn ConsensusEnclave + Send + Sync>,
+    authenticator: Arc<dyn Authenticator + Send + Sync>,
     logger: Logger,
     sessions: Arc<Mutex<HashSet<S>>>,
 }
 
-impl<E: ConsensusEnclave + Send, S: Session> AttestedApiService<E, S> {
-    pub fn new(enclave: E, logger: Logger) -> Self {
+impl<S: Session> AttestedApiService<S> {
+    pub fn new(
+        enclave: Arc<dyn ConsensusEnclave + Send + Sync>,
+        authenticator: Arc<dyn Authenticator + Send + Sync>,
+        logger: Logger,
+    ) -> Self {
         Self {
             enclave,
+            authenticator,
             logger,
             sessions: Arc::new(Mutex::new(HashSet::default())),
         }
     }
 }
 
-impl<E: ConsensusEnclave + Send> AttestedApi for AttestedApiService<E, PeerSession> {
+impl AttestedApi for AttestedApiService<PeerSession> {
     fn auth(&mut self, ctx: RpcContext, request: AuthMessage, sink: UnarySink<AuthMessage>) {
         let _timer = SVC_COUNTERS.req(&ctx);
         mc_common::logger::scoped_global_logger(&rpc_logger(&ctx, &self.logger), |logger| {
+            if let Err(err) = self.authenticator.authenticate_rpc(&ctx) {
+                return send_result(ctx, sink, err.into(), &logger);
+            }
```

Comment:
This is the pattern we need to follow when we want to require RPC calls to go through authentication.